services/core-api/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -178,6 +173,21 @@ def post(self, mine_guid):
                     rec_string, mine_incident_id=incident.mine_incident_id)
                 new_recommendation.save()
 
+        categories = data.get('categories')
+        if categories is not None:
+            for category in categories:
+                if category is None:
+                    continue
+                new_category = MineIncidentCategoryXref(
+                    mine_incident_id=incident.mine_incident_id,
+                    mine_incident_category_code=category)
+                incident.categories.append(new_category)
+
+        try:
+            incident.save()
+        except Exception as e:
+            raise InternalServerError(f'Error when saving: {e}')
```

Comment:
I moved this to the bottom of the POST function because it makes more sense there.